src/ir/annotations.rs

```diff
@@ -0,0 +1,119 @@
+use clang;
+
+#[derive(Copy, PartialEq, Clone, Debug)]
+pub enum FieldAccessorKind {
+    None,
+    Regular,
+    Unsafe,
+    Immutable,
+}
+
+/// Annotations for a given item, or a field.
+#[derive(Clone, PartialEq, Debug)]
+pub struct Annotations {
+    /// Whether this item is marked as opaque. Only applies to types.
+    opaque: bool,
+    /// Whether this item should be hidden from the output. Only applies to
+    /// types.
+    hide: bool,
+    /// Whether this type should be replaced by another. The name must be the
+    /// canonical name that that type would get.
+    use_as: Option<String>,
+    /// Manually disable deriving copy/clone on this type. Only applies to
+    /// struct or union types.
+    no_copy: bool,
+    /// Whether fields should be marked as private or not. You can set this on
+    /// structs (it will apply to all the fields), or individual fields.
+    private_fields: Option<bool>,
+    /// The kind of accessor this field will have. Also can be applied to
+    /// structs so all the fields inside share it by default.
+    accessor_kind: Option<FieldAccessorKind>,
+}
+
+fn parse_accessor(s: &str) -> FieldAccessorKind {
+    match s {
+        "false" => FieldAccessorKind::None,
+        "unsafe" => FieldAccessorKind::Unsafe,
+        "immutable" => FieldAccessorKind::Immutable,
+        _ => FieldAccessorKind::Regular,
+    }
+}
+
+impl Default for Annotations {
+    fn default() -> Self {
+        Annotations {
+            opaque: false,
+            hide: false,
+            use_as: None,
+            no_copy: false,
+            private_fields: None,
+            accessor_kind: None
+        }
+    }
+}
+
+impl Annotations {
+    pub fn new(cursor: &clang::Cursor) -> Option<Annotations> {
+        let mut anno = Annotations::default();
+        let mut matched_one = false;
+        anno.parse(&cursor.comment(), &mut matched_one);
+
+        if matched_one {
+            Some(anno)
+        } else {
+            None
+        }
+    }
+
+    pub fn hide(&self) -> bool {
+        self.hide
+    }
+
+    pub fn opaque(&self) -> bool {
+        self.opaque
+    }
+
+    pub fn use_as(&self) -> Option<&str> {
```

Comment:
This is used to implement the `replaces` annotation. I've given it a more meaningful name and a doc comment :)